### PR TITLE
Added error reporting for sources

### PR DESCRIPTION
**Pull request by theduckylittle**
Creates a new `sourceErrors` object on the
MapInfoReducer. `sourceErrors` are reset everytime
the map is moved.

### CODE REVIEW

File: __tests__/reducers/mapinfo.test.js [hunk 1]
```diff
@@ -82,4 +83,39 @@ describe('mapinfo reducer', () => {
...
+    };
+
+    expect(reducer(state, action)).toEqual({
+      sourceErrors: {
+        test: true,
+      },
+    });
+  });
+
+  it('should errored sources', () => {
```
Comment (bartvde) [inline __tests__/reducers/mapinfo.test.js]: clear missing here?

File: src/action-types.js [hunk 2]
```diff
@@ -86,6 +86,7 @@ export const MAPINFO = {
   SET_RESOLUTION: 'MAPINFO.SET_RESOLUTION',
   SET_PROJECTION: 'MAPINFO.SET_PROJECTION',
   REQUEST_REDRAW: 'MAPINFO.REQUEST_REDRAW',
+  SET_SOURCE_ERROR: '',
```
Comment (bartvde) [inline src/action-types.js]: I guess this is not intentionally set as an empty string?

Comment (bartvde) [reply src/action-types.js]: also, is the clear source errors const missing?

File: src/actions/mapinfo.js [hunk 3]
```diff
@@ -88,3 +88,26 @@ export function requestRedraw() {
     type: MAPINFO.REQUEST_REDRAW,
   };
 }
+
+/** Set an error for a source.
+ *
+ *  @param {String} srcName - The name of the source with the error
+ *
+ *  @returns {Object} Action object to request a map redraw.
```
Comment (bartvde) [inline src/actions/mapinfo.js]: copy paste?

File: src/actions/mapinfo.js [hunk 4]
```diff
@@ -88,3 +88,26 @@ export function requestRedraw() {
...
+export function setSourceError(srcName) {
+  return {
+    type: MAPINFO.SET_SOURCE_ERROR,
+    srcName,
+  };
+}
+
+/** Clear the source errors
+ *
+ *  @returns {Object} Action object to request a map redraw.
```
Comment (bartvde) [inline src/actions/mapinfo.js]: same here, copy paste?

File: src/components/map.js [hunk 5]
```diff
@@ -370,6 +370,8 @@ function getLoaderFunction(glSource, mapProjection, baseUrl) {
           this.addFeatures(GEOJSON_FORMAT.readFeatures(features, readFeatureOpt));
         }
       }).catch((error) => {
+        // use the event name tileloaderror for consistency.
+        this.dispatchEvent('tileloaderror');
```
Comment (bartvde) [inline src/components/map.js]: just checking, but this is the source scope here?

Comment (bartvde) [reply src/components/map.js]: I feel a little uncomfortable dispatching an event here on the openlayers object, but I also see in this function we don't have the map component scope to trigger the action directly

Comment (theduckylittle) [reply src/components/map.js]: This is necessary to "fake" the same behaviour that is available from the tile layers.  it also provides for easier testing as we have better test cases for the GeoJson type layers.

**Pull request merged**